Document how to use varargs in Argument Clinic (#1277)
python/devguide: development-tools/clinic.rst

a *type* argument to the object converter for :c:macro:`METH_O`.

How to convert var-positional parameter functions
-------------------------------------------------

To convert a var-positional parameter function,
prepending the parameter name with ``*`` ,

> and the parameter should use the ``object`` converter::

Reviewer: What happens if I use something else? E.g., what if I want to accept an arbitrary number of ints?

Also, what type does the implementation function receive?

Author: Thank you for the review! I have added a new paragraph to explain more about it.

a *type* argument to the object converter for :c:macro:`METH_O`.

How to convert var-positional parameter functions
-------------------------------------------------

To convert a var-positional parameter function,
prepending the parameter name with ``*`` ,
and the parameter should use the ``object`` converter::

   /*[clinic input]
   var_positional_sample

       foo: int
       *args: object
   [clinic start generated code]*/

> The implementation function will receive var-positional arguments
> as a tuple, you can either use the tuple directly
> or parsing them into C types using :c:func:`!PyArg_Parse*` functions.

Reviewer 2: ```suggestion
The implementation function will receive var-positional arguments
as a tuple; you can either use the tuple directly or parse it into
C types via :c:func:`PyArg_ParseTuple` or :c:func:`PyArg_UnpackTuple`.
```

Am I right concerning the functions to use? I don't think we have other
functions that would be relevant in this specific case.

Reviewer: I think it's OK to skip this part. It's important to mention that it's a tuple; for figuring out what to do with the tuple we can point to the C API documentation.
```suggestion
The implementation function will receive var-positional arguments
as a tuple.
```

Reviewer 2: I'm also ok if we omit the C API mention!

> .. versionadded:: 3.11

Reviewer 2: Argument Clinic is somewhat an internal tool so I'm not sure we should mention when it was added, or do we?

Reviewer 3: Since this is the devguide @picnixz, I think it is fine to mention or leave out.

Reviewer 2: I'll leave it at aisk's discretion!
